AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift

```diff
@@ -151,17 +148,27 @@
                     self.loadingNavigation = wkWebView.loadHTMLString(self.payload, baseURL: Bundle.main.bundleURL)
                 }
 
-                // only show the message if the monitor allows it
-                guard self.messageMonitor.show(message: self) else {
-                    self.listener?.onShowFailure()
-                    return
+                // get off main thread while delegate has control to prevent pause on main thread
+                DispatchQueue.global().async {
+                    self.handleShouldShow(self, webview: wkWebView, delegateControl: delegateControl)
                 }
+            }
+        }
 
+        private func handleShouldShow(_ message: FullscreenMessage, webview: WKWebView, delegateControl: Bool) {
+            // only show the message if the monitor allows it
+            guard message.messageMonitor.show(message: message, delegateControl: delegateControl) else {
+                message.listener?.onShowFailure()
+                return
+            }
+
+            // dispatch UI activity back to main thread
+            DispatchQueue.main.async {
                 // notify global listeners
-                self.listener?.onShow(message: self)
-                self.messagingDelegate?.onShow(message: self)
+                message.listener?.onShow(message: message)
+                message.messagingDelegate?.onShow(message: message)
```

Comment:
This is still customer controlled code on main thread (potentially) ?

It's confusing, but should we consider dispatching this off to global as well?  would that cause a problem if listener.onshow and messagingDelegate.onShow happened non-linearly (or -- more specifically -- farther apart in a non blocking fashion before message.displayWithAnimation ?)